File: ObjectDT/contours.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
#include <vector>

using namespace cv;
using namespace std;

int main(int argc, char **argv){
    if(argc < 2){
        cerr <<"Send image"<< endl;
        return EXIT_FAILURE;
    }

    Mat image = imread(argv[1]);
    if(image.empty()){
        cerr <<"Fail to load image"<< endl;
        return EXIT_FAILURE;
    }

    Mat gray;
    cvtColor(image, gray, COLOR_BGR2GRAY);

    Mat binary;
    threshold(gray, binary, 127, 255, THRESH_BINARY);

    vector<vector<Point>> contours;
    vector<Vec4i> hierarchy;
    findContours(binary, contours, hierarchy, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);

    Mat contourImage = image.clone();
    drawContours(contourImage, contours, -1, Scalar(0, 255, 0), 2);

    imshow("Original Image", image);
    imshow("Contours", contourImage);

    imwrite("countour.jpg", contourImage); //save image
    
    waitKey(0);
    destroyAllWindows();

    return EXIT_SUCCESS;
}

File: ObjectDT/contours.py
import cv2 

path_img = "../Data/Perseus.jpg"
img = cv2.imread(path_img)

#Bin image
gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
thresh = 150
maxval = 255
_, thres = cv2.threshold(gray, thresh, maxval, cv2.THRESH_BINARY_INV + cv2.THRESH_OTSU)

#Find contours
contour, _ = cv2.findContours(thres, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)

cv2.drawContours(img, contour, -1, (0, 255, 0), 2) 

#show
#img_resized = cv2.resize(img, (640, 480)) #to show better
cv2.imshow("Contours", img)
cv2.waitKey(0)
cv2.destroyAllWindows()